app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+import json
+from decimal import Decimal
+
+
+def calculate_profit(input_filename: str) -> None:
+    with open(input_filename, "r") as input_file:
+        trades = json.load(input_file)
+
+    earned_money = Decimal("0")
+    matecoin_account = Decimal("0")
+
+    for trade in trades:
+        bought = Decimal(trade.get("bought") or 0)
+        sold = Decimal(trade.get("sold") or 0)
+        matecoin_price = Decimal(trade["matecoin_price"])
+
+        if bought > 0:
+            earned_money -= bought * matecoin_price
+            matecoin_account += bought
+        elif sold > 0:
```

Comment:
You should not assume that only `bought` or `sold` operation is possible within one trade. Replace `elif` with `if` on this line.